### PR TITLE
Add restart command.

### LINKED ISSUE #46: Command to restart hie
Due to some reason (probably all the way down in the GHC API) pretty much every Haskell IDE tool consumes a ton of memory and the amount usually increases with use. I currently solve this by monitoring memory usage and restarting VS Code when I see it's taking up too much. The problem is that this kills terminal sessions in the editor.

I'd like to be able to restart just the hie backend.

### PR DESCRIPTION
**Pull request by gdziadkiewicz**
Fix #46

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -196,6 +224,11 @@
         "command": "hie.commands.importIdentifier",
         "title": "Haskell: Import identifier",
         "description": "Imports a function or type based on a Hoogle search"
+      },
+      {
+        "command": "hie.commands.restartHie",
+        "title": "Haskell: Restart HIE",
+        "description": "Restart the Hie LPS server"
```
Comment (fendor) [inline package.json]: typo, should be LSP.

Comment (gdziadkiewicz) [reply package.json]: Fixed.

**Comment by alanz** [comment]
Please resolve the conflict, then we can see about merging and releasing

**Comment by alanz** [comment]
How does the restart work? Is it activated via the Command Palette?  Because when it is done you should get the message from hie giving its version

**Comment by gdziadkiewicz** [comment]
It stops and then starts each of the lang clients in the path to lang
client dict.Yes, it is activated from the command palette. I can see the
death of the hie instance and creation of new hie instance in the output
window of vs code and in taskmgr. I will record it as gif and upload it
here together with the output to show what can be expected after this
change.

I only tested it on Windows 10. Could someone else give it a test drive on
other configuration?

**Comment by gdziadkiewicz** [comment]
restart
[output.log](https://github.com/alanz/vscode-hie-server/files/4138513/output.log)

**Comment by jneira** [comment]
@gdziadkiewicz thanks for the pr! we'll try to do a release asap

**Pull request merged**